src/test/scala/com/mwz/sonar/scala/scalastyle/ScalastyleRulesRepositoryTest.scala

```diff
@@ -0,0 +1,229 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala
+package scalastyle
+
+import org.scalastyle._
+import org.scalatest.{FlatSpec, Inspectors, LoneElement, Matchers}
+import org.sonar.api.batch.rule.Severity
+import org.sonar.api.rule.RuleStatus
+import org.sonar.api.rules.RuleType
+import org.sonar.api.server.rule.RuleParamType
+import org.sonar.api.server.rule.RulesDefinition.{Context, Repository}
+
+class ScalastyleRulesRepositoryTest extends FlatSpec with Matchers with Inspectors with LoneElement {
+
+  trait Ctx {
+    val context = new Context()
+    new ScalastyleRulesRepository().define(context)
+    val repository: Repository = context.repositories.loneElement
+  }
+
+  "ScalastyleRulesRepository" should "define rules repository" in new Ctx {
+    context.repositories().size shouldBe 1
+  }
+
+  it should "correctly define repository properties" in new Ctx {
+    repository.key shouldBe ScalastyleRulesRepository.RepositoryKey
+    repository.name shouldBe ScalastyleRulesRepository.RepositoryName
+    repository.language shouldBe Scala.LanguageKey
+  }
+
+  it should "include all Scalastyle inspections" in new Ctx {
+    repository.rules.size should be > 0
```

Comment:
So this is just to make sure that the generated collection is not empty - if for some reason this was the case, there would be no rules in the repository, but the test would pass without this check because the `repository.rules.size == ScalastyleInspections.AllInspections.size == 0`.

Reply:
That makes sense, however I think it would be better if we check for that in a test suit for the generated file itself. For example we could check that the collection size is equal to [69](http://www.scalastyle.org/rules-1.0.0.html).

Reply:
I see what you mean, although to check the length of the collection that substituted the `???` in the template file, you actually need to get the generated file compiled to be sure that scala meta did the transformation and the file was saved, so it's much easier to write a test around it in the main project where the file is actually part of the sources so you can refer to the collection.

I'm also a bit reluctant to hardcode the size of the collection in the tests, as any time an inspection would be added or deleted, the tests would start to fail.

Reply:
> " so it's much easier to write a test around it in the main project where the file is actually part of the sources so you can refer to the collection."

Oh yeah absolutely, I was refering to a test suit in the main project.
That way we will be testing the generated file itself, instead of the logic that generated the file.

> "I'm also a bit reluctant to hardcode the size of the collection in the tests, as any time an inspection would be added or deleted, the tests would start to fail."

Well, I also find anoying to change the tests everytime the number of inspections change. However, that number will only change whenever we change the version of Scalastyle _(or Scapegoat, I'm planning to do something silimir with it)_ that we use. And, in that way, I think is a good idea to check against a harcoded number of inspections, becasue we should be sure that the collection contains all inspections of the specifict version we are using.

Reply:
Ok, you convinced me on that one, I'll make the change.